advent-of-code/rucksack-problem/src/rucksack_problem/core.clj

```diff
@@ -0,0 +1,136 @@
+;; Advent of Code Day 03
+
+;; Part 1
+;;      Process a single rucksack
+;;   1. Divide rucksack string into 2 halves
+;;   2. Check for the common item (only one exists) in both halves
+;;   3. Find the priority of the common item using the below relationships:
+;;      Priority of a - z: 1 - 26
+;;      Priority of A - Z: 27 - 52)
+;;
+;;      And then expand to multiple rucksacks to solve Part 1
+
+;; Lifting some input restrictions as another exercise
+;; Lift restriction of a single common item. Code generic solution where priorities are counted
+;; for each pair of common items between the 2 compartments.
+;; 1. create hashmap for compartment 1 and 2 with key: item, value: frequency of items
+;; 2. find pairs of common keys between both hashmaps
+;; 3. minimum value of each key pair is the priority multiple
+;; 4. sum of (priority * multiple) for each key pair
+
+;; Part 2
+;; For 1 group of 3 rucksacks
+;; 1. find common item in each group of 3 rucksacks.
+;; 2. find priority of common item
+;; Repeat the above for each group of 3 rucksacks, and sum the priorities of common items.
+
+(ns rucksack-problem.core
+  (:require [clojure.string :as string]
+            [clojure.set :as set])
+  (:gen-class))
+
+(defn input-file->string-seq
+  "Read file and extract strings separated by \n into a vector of strings"
+  [f]
+  (-> (slurp f)
+      (string/split #"\n")))
+
+(defn compartmentalize
+  "Divide rucksack contents into 2 compartments
+   1 string -> 2 lists of chars"
+  [rucksack-contents]
+  (let [half-length (/ (count rucksack-contents) 2)]
+    (->> rucksack-contents
+         (partition half-length))))
```

Comment:
[suggestion] Consider a more precise name, like `partition-half`.

A good heuristic for checking if the name is good: Can you guess what this function does if you were only given its name (and no access to the docstring or implementation)?

Applying this heuristic here:
* `compartmentalize` tells us that I can give it a rucksack and I get something that arranges its items into compartments. But it doesn't tell us how many compartments and criteria for which it makes them.

Reply:
Agreed, `partition-half` is very clear. 
I've been having trouble deciding when to use words from the problem set (compartmentalize) vs. words that more clearly describe what the function does. Will now tend towards the latter, that is more descriptive function names.